fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/ReorderJoin.java

```diff
@@ -76,28 +75,31 @@
 public class ReorderJoin extends OneRewriteRuleFactory {
     @Override
     public Rule build() {
-        return logicalFilter(subTree(LogicalJoin.class, LogicalFilter.class)).thenApply(ctx -> {
-            if (ctx.statementContext.getConnectContext().getSessionVariable().isDisableJoinReorder()) {
-                return null;
-            }
-            LogicalFilter<Plan> filter = ctx.root;
-
-            Map<Plan, JoinHintTypeAndMarkJoinSlot> planToJoinMembers = Maps.newHashMap();
-            Plan plan = joinToMultiJoin(filter, planToJoinMembers);
-            Preconditions.checkState(plan instanceof MultiJoin);
-            MultiJoin multiJoin = (MultiJoin) plan;
-            ctx.statementContext.setMaxNArayInnerJoin(multiJoin.children().size());
-            Plan after = multiJoinToJoin(multiJoin, planToJoinMembers);
-            return after;
-        }).toRule(RuleType.REORDER_JOIN);
+        return logicalFilter(subTree(LogicalJoin.class, LogicalFilter.class))
+            .whenNot(filter -> filter.child() instanceof LogicalJoin
+                    && ((LogicalJoin<?, ?>) filter.child()).isMarkJoin())
```

Comment:
if the plan is
```
Filter()
+---Join(non mark join)
    +---Join(mark join)
```

what will happen? we always has `Project` on the top of `Filter(Mark Join)`? right?